ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+
+The usual and valid critique of the Actor Model, in particular in Akka, is that due to its dynamic nature and due to this single message entry point into an Actor’s behavior we forego all static type checking of Actor interactions—the ActorRef excepts `Any` message and the behavior is a function from `Any` to `Unit`. This is however not a fundamental restriction as the nascent [Akka Typed](http://doc.akka.io/docs/akka/snapshot/scala/typed.html) project shows: while it took a few tries to arrive at a formulation that works, we now have demonstrated that fully type-safe Actor interactions are possible without performance overhead and very little syntactic difference (with Scala’s case classes; in Java defining the required message classes involves more boilerplate). Using this new model we can transform a classical interface into an Actor definition by changing method declarations into class definitions—keeping the arguments as fields—and adding a reply-to ActorRef in order to transmit the return value if there is one; then the role of the `.` method selection is taken by the `!` message send operator. For this reason it might be time to rethink the valuation of classical interfaces and the Actor Model.
+
+## Closing Remarks
+
+This investigation has been sparked by the question of whether Akka.NET shall implement the «Virtual Actor» abstraction that Orleans offers, in addition to the classical Actor Model copied from Akka. After having seen the differences in goals and philosophy I would like to suggest that Orleans use the terms «Grain» and «Silo» to describe the undoubtedly useful abstraction that it provides, but refrain from using «Virtual Actor» terminology for two reasons:
+
+  * Orleans’ explicit goal is to hide the Actor-inspired implementation details and offer a hassle-free “elastic object” programming model that does not require distributed programming expertise.
```

Comment:
Thank you! I just pushed updates to those points where I think we reached agreement, feel free to review and ask more. This is indeed a rather useful exercise.